### PR TITLE
103 create app settings page

### LINKED ISSUE #103: Create app Settings page
Add some settings to the app
- ability to remove some data from the data stores to reduce memory usage
- notification settings
- dark theme
- text size

### PR DESCRIPTION
**Pull request by weihanli101**
this closes #103 

- Creates the UI  for settings page
- Added dynamic theme switching for the application

Warnings:
- the add comments fragment's background does not follow the set theme. Have no idea why.

### CODE REVIEW

File: app/src/main/java/ca/marshallasch/veil/FragmentSettings.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        return view;
+    }
+
+    private void toggleDarkTheme(boolean isDarkTheme){
+        SharedPreferences preferences = getActivity().
+                getSharedPreferences(MainActivity.SYSTEM_PREF, getActivity().MODE_PRIVATE);
+        SharedPreferences.Editor editor = preferences.edit();
+        editor.putBoolean(IS_DARK_THEME_TOGGLED, isDarkTheme);
+        editor.apply();
+        Intent intent = new Intent(getActivity(), MainActivity.class);
```
[the remark this answers is not in the document]
Comment (MarshallAsch) [reply app/src/main/java/ca/marshallasch/veil/FragmentSettings.java]: fixed in 9662c12

File: app/src/main/java/ca/marshallasch/veil/FragmentSettings.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.fragment_settings, container, false);
+
+        //set dark theme toggle to save preference
+        Switch darkThemeToggle = view.findViewById(R.id.toggle_dark_theme);
+        SharedPreferences preferences = getActivity().
+                getSharedPreferences(MainActivity.SYSTEM_PREF, getActivity().MODE_PRIVATE);
```
[the remark this answers is not in the document]
Comment (MarshallAsch) [reply app/src/main/java/ca/marshallasch/veil/FragmentSettings.java]: fixed in 9662c12

File: app/src/main/java/ca/marshallasch/veil/MainActivity.java [hunk 3]
```diff
@@ -63,10 +63,16 @@ public void onServiceDisconnected(ComponentName componentName) {
     };
 
 
-
     @Override
     protected void onCreate(Bundle savedInstanceState) {
-        setTheme(R.style.AppTheme);
+        SharedPreferences preferences = getSharedPreferences(SYSTEM_PREF, MODE_PRIVATE);
```
[the remark this answers is not in the document]
Comment (weihanli101) [reply app/src/main/java/ca/marshallasch/veil/MainActivity.java]: Default shared preferences points to a file. getSharedPreferences returns an instance

File: app/src/main/java/ca/marshallasch/veil/MainActivity.java [hunk 4]
```diff
@@ -174,8 +181,7 @@ public boolean onOptionsItemSelected(MenuItem item)
         return true;
     }
 
-    /**
-     * Navigate to the given fragment.
+    /** Navigate to the given fragment.
```
[the remark this answers is not in the document]
Comment (weihanli101) [reply app/src/main/java/ca/marshallasch/veil/MainActivity.java]: just a mistake.

File: app/src/main/res/layout/comment_item.xml [hunk 5]
```diff
@@ -31,6 +31,7 @@
         android:layout_marginStart="8dp"
         android:layout_marginTop="8dp"
         android:text="@string/author_name_place_holder"
+        android:textColor="?themeAccent"
```
[the remark this answers is not in the document]
Comment (weihanli101) [reply app/src/main/res/layout/comment_item.xml]: We have different colours for our textviews. Ex. subtitles and titles. We can't generalize textView.

File: app/src/main/res/layout/fragment_create_post.xml [hunk 6]
```diff
@@ -86,8 +86,8 @@
                     android:maxLines="1"/>
             </android.support.design.widget.TextInputLayout>
 
-            <android.support.design.widget.TextInputLayout
-                android:id="@+id/tags_text_input"
+            <Spinner
```
[the remark this answers is not in the document]
Comment (weihanli101) [reply app/src/main/res/layout/fragment_create_post.xml]: On the current build the spinner is white on the dark theme and black on the white theme? Is this something that still needs to be changed?

File: app/src/main/res/menu/main_menu.xml [hunk 7]
```diff
@@ -1,12 +1,14 @@
...
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <item
+        android:id="@+id/app_settings"
+        android:title="@string/app_settings"/>
 
     <item
         android:id="@+id/setup"
-        android:title="@string/setup_right_mesh"/>
+        android:title="@string/rightmesh_settings"/>
```
[the remark this answers is not in the document]
Comment (weihanli101) [reply app/src/main/res/menu/main_menu.xml]: screen shot 2018-07-26 at 1 50 39 pm
Was this change fixed? Doesn't seem to be on the version I pulled.

File: app/src/main/java/ca/marshallasch/veil/MainActivity.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (MarshallAsch) [reply app/src/main/java/ca/marshallasch/veil/MainActivity.java]: `getDefaultSharedPreferences(Context context)` also gets an instance. read the doc I sent you.

File: app/src/main/res/layout/comment_item.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (MarshallAsch) [reply app/src/main/res/layout/comment_item.xml]: int the parent theme you should be able to set what the different textview themes are and have those settings propagate without having to actually change the individual textboxes.

File: app/src/main/res/menu/main_menu.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (MarshallAsch) [reply app/src/main/res/menu/main_menu.xml]: Yes I fixed it. you created a pointless resource tho. The label was already in a string resource and you just made another resource for the title.

File: app/src/main/res/layout/fragment_create_post.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (MarshallAsch) [reply app/src/main/res/layout/fragment_create_post.xml]: I never set any theme things on the spinners or on anything that was added from the other branches.

**Comment by MarshallAsch** [comment]
Please check to make sure that you can not just override the material button and text view themes in the app theme. You should not need to set the fields in every single button and text view for the font colors and background colors.

**Pull request merged**